hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -1448,11 +1448,12 @@ private HoodieData<HoodieRecord> getRecordIndexUpdates(HoodieData<WriteStatus> w
                 LOG.error(msg);
                 throw new HoodieMetadataException(msg);
               }
+              // for updates, we can skip updating RLI partition in MDT
+            } else {
+              hoodieRecord = HoodieMetadataPayload.createRecordIndexUpdate(
```

Comment:
so newLocation is present but currentLocation is not.. this should happen only with append handle right and no other place? I assume you have already checked that

Reply:
nope. even w/ merge handle. essentially, if a record is an update record both locations will be set. if not, only new location will be set.